File: src/pyWATS/modules/workflow.py
"""
Workflow module for WATS API.

This module provides functionality for managing workflows, steps,
and workflow-related operations in the WATS system.
"""

from typing import List, Optional, Dict, Any
from enum import Enum
from .base import BaseModule
from ..exceptions import WATSException, WATSNotFoundError


class StatusEnum(Enum):
    """Status enumeration for workflow definitions."""
    RELEASED = "Released"
    DRAFT = "Draft"
    OBSOLETE = "Obsolete"


class ActivityTestResult(Enum):
    """Activity test result enumeration."""
    PASSED = "Passed"
    FAILED = "Failed"
    TERMINATED = "Terminated"
    ERROR = "Error"


class ActivityMethod(Enum):
    """Activity method enumeration."""
    AUTOMATIC = "Automatic"
    MANUAL = "Manual"
    SEMI_AUTOMATIC = "SemiAutomatic"


class WorkflowResponse:
    """Response object for workflow operations."""
    
    def __init__(self, success: bool = True, message: str = "", data: Any = None):
        self.success = success
        self.message = message
        self.data = data
        self.workflow_state = None
        self.next_activity = None
        self.validation_results = []


class WorkflowModule(BaseModule):
    """
    Workflow and step management module.
    
    Provides methods for:
    - Test workflow management (start/end test)
    - Workflow lifecycle operations (initialize, check in/out)
    - Validation and user input handling
    - Repair workflow management
    - Unit workflow operations (add/remove units)
    - Workflow state management (suspend/resume/cancel)
    """

    def is_connected(self) -> bool:
        """Check if workflow module is connected."""
        raise NotImplementedError("Workflow.is_connected not implemented")

    def start_test(self, serial_number: str, part_number: str, operation: str,
                  input_values: Dict[str, Any], prompt_operator: bool = False,
                  always_on_top: bool = True, workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Start a test workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.start_test not implemented")

    def end_test(self, serial_number: str, part_number: str, operation: str,
                result: ActivityTestResult, force_exit: bool, input_values: Dict[str, Any],
                prompt_operator: bool = False, always_on_top: bool = True,
                workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        End a test workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            result: Test result
            force_exit: Force exit flag
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.end_test not implemented")

    def validate(self, serial_number: str, part_number: str, method: ActivityMethod,
                name: str, input_values: Dict[str, Any],
                workflow_definition_status: StatusEnum = StatusEnum.RELEASED,
                generate_image: bool = False) -> WorkflowResponse:
        """
        Validate workflow activity.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            method: Activity method
            name: Activity name
            input_values: Input values dictionary
            workflow_definition_status: Workflow definition status
            generate_image: Generate image flag
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.validate not implemented")

    def initialize(self, serial_number: str, part_number: str, input_values: Dict[str, Any],
                  prompt_operator: bool = False, always_on_top: bool = True,
                  workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Initialize workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.initialize not implemented")

    def check_in(self, serial_number: str, part_number: str, operation: str,
                input_values: Dict[str, Any], prompt_operator: bool = False,
                always_on_top: bool = True, workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Check in unit to workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.check_in not implemented")

    def check_out(self, serial_number: str, part_number: str, operation: str,
                 input_values: Dict[str, Any], prompt_operator: bool = False,
                 always_on_top: bool = True, workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Check out unit from workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.check_out not implemented")

    def user_input(self, serial_number: str, part_number: str, operation: str,
                  user_input: str, input_values: Dict[str, Any],
                  prompt_operator: bool = False, always_on_top: bool = True,
                  workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Handle user input in workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            user_input: User input string
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.user_input not implemented")

    def start_repair(self, serial_number: str, part_number: str, operation: str,
                    user_input: str, input_values: Dict[str, Any],
                    prompt_operator: bool = False, always_on_top: bool = True,
                    workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Start repair workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            user_input: User input string
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.start_repair not implemented")

    def end_repair(self, serial_number: str, part_number: str, operation: str,
                  user_input: str, input_values: Dict[str, Any],
                  prompt_operator: bool = False, always_on_top: bool = True,
                  workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        End repair workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            user_input: User input string
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.end_repair not implemented")

    def scrap(self, serial_number: str, part_number: str, operation: str,
             user_input: str, input_values: Dict[str, Any],
             prompt_operator: bool = False, always_on_top: bool = True,
             workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Scrap unit in workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            operation: Operation name
            user_input: User input string
            input_values: Input values dictionary
            prompt_operator: Prompt operator flag
            always_on_top: Always on top flag
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.scrap not implemented")

    def suspend(self, serial_number: str, part_number: str,
               workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Suspend workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.suspend not implemented")

    def resume(self, serial_number: str, part_number: str,
              workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Resume workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.resume not implemented")

    def cancel(self, serial_number: str, part_number: str,
              workflow_definition_status: StatusEnum = StatusEnum.RELEASED) -> WorkflowResponse:
        """
        Cancel workflow.
        
        Args:
            serial_number: Serial number of the unit
            part_number: Part number of the unit
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.cancel not implemented")

    def add_unit(self, serial_number: str, part_number: str, child_serial_number: str,
                child_part_number: str, activity_name: str, input_values: Dict[str, Any],
                workflow_definition_status: StatusEnum) -> WorkflowResponse:
        """
        Add unit to workflow.
        
        Args:
            serial_number: Serial number of the parent unit
            part_number: Part number of the parent unit
            child_serial_number: Serial number of the child unit
            child_part_number: Part number of the child unit
            activity_name: Activity name
            input_values: Input values dictionary
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.add_unit not implemented")

    def remove_unit(self, serial_number: str, part_number: str, child_serial_number: str,
                   child_part_number: str, activity_name: str, input_values: Dict[str, Any],
                   workflow_definition_status: StatusEnum) -> WorkflowResponse:
        """
        Remove unit from workflow.
        
        Args:
            serial_number: Serial number of the parent unit
            part_number: Part number of the parent unit
            child_serial_number: Serial number of the child unit
            child_part_number: Part number of the child unit
            activity_name: Activity name
            input_values: Input values dictionary
            workflow_definition_status: Workflow definition status
            
        Returns:
            WorkflowResponse object
        """
        raise NotImplementedError("Workflow.remove_unit not implemented")

    # Legacy methods for backward compatibility
    def get_all(self) -> List[Dict[str, Any]]:
        """Get all workflows."""
        return [{"message": "Workflow functionality will be implemented with actual API endpoints"}]
    
    def get_by_id(self, workflow_id: str) -> Dict[str, Any]:
        """Get a specific workflow by ID."""
        self._validate_id(workflow_id, "workflow")
        return {"message": f"Workflow {workflow_id} functionality will be implemented with actual API endpoints"}
    
    def get_steps(self, workflow_id: str) -> List[Dict[str, Any]]:
        """Get steps for a specific workflow."""
        self._validate_id(workflow_id, "workflow")
        return [{"message": f"Steps for workflow {workflow_id} will be implemented with actual API endpoints"}]

File: referenced_code/Interface.MES/Workflow/Workflow.cs
extern alias newclientapi;

using napi = newclientapi::Virinco.WATS.Interface.MES;
using napict = newclientapi::Virinco.WATS.Service.MES.Contract;
using Virinco.WATS.Service.MES.Contract;
using System.Collections.Generic;

namespace Virinco.WATS.Interface.MES.Workflow
{
    /// <summary>
    /// Class that represent a workflow
    /// </summary>
    public class Workflow : MesBase
    {
        private napi.Workflow.Workflow _instance;

        internal Workflow(napi.Workflow.Workflow workflow)
        {
            this._instance = workflow;
        }

        /// <summary>
        /// True if connected to server
        /// </summary>
        /// <returns></returns>
        new public bool isConnected() => _instance.isConnected();

        /// <summary>
        /// Start automated test
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse StartTest(string SerialNumber, string PartNumber, string Operation, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.StartTest(SerialNumber, PartNumber, Operation, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// End test with result
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="Result"></param>
        /// <param name="ForceExit"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse EndTest(string SerialNumber, string PartNumber, string Operation, ActivityTestResult Result, bool ForceExit, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.EndTest(SerialNumber, PartNumber, Operation, (napict.ActivityTestResult)(int)Result, ForceExit, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Validate workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Method"></param>
        /// <param name="Name"></param>
        /// <param name="inputValues"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <param name="generateImage"></param>
        /// <returns></returns>
        public WorkflowResponse Validate(string SerialNumber, string PartNumber, ActivityMethod Method, string Name, Dictionary<string, object> inputValues, StatusEnum workflowDefinitionStatus = StatusEnum.Released, bool generateImage = false)
            => new WorkflowResponse(_instance.Validate(SerialNumber, PartNumber, (napict.ActivityMethod)(int)Method, Name, inputValues, (napict.StatusEnum)(int)workflowDefinitionStatus, generateImage));

        /// <summary>
        /// Initialize a workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse Initialize(string SerialNumber, string PartNumber, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.Initialize(SerialNumber, PartNumber, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Performs a Check-In on a workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse CheckIn(string SerialNumber, string PartNumber, string Operation, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.CheckIn(SerialNumber, PartNumber, Operation, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Performs a Check-Out on a workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse CheckOut(string SerialNumber, string PartNumber, string Operation, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.CheckOut(SerialNumber, PartNumber, Operation, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Performs a UserInput on a workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="UserInput"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse UserInput(string SerialNumber, string PartNumber, string Operation, string UserInput, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.UserInput(SerialNumber, PartNumber, Operation, UserInput, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Start repair
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="UserInput"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse StartRepair(string SerialNumber, string PartNumber, string Operation, string UserInput, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.StartRepair(SerialNumber, PartNumber, Operation, UserInput, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// End repair
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="UserInput"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse EndRepair(string SerialNumber, string PartNumber, string Operation, string UserInput, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.EndRepair(SerialNumber, PartNumber, Operation, UserInput, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Scrap a unit under repair
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="Operation"></param>
        /// <param name="UserInput"></param>
        /// <param name="inputValues"></param>
        /// <param name="promptOperator"></param>
        /// <param name="AlwaysOnTop"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse Scrap(string SerialNumber, string PartNumber, string Operation, string UserInput, Dictionary<string, object> inputValues, bool promptOperator = false, bool AlwaysOnTop = true, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.Scrap(SerialNumber, PartNumber, Operation, UserInput, inputValues, promptOperator, AlwaysOnTop, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Suspends a workflow 
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse Suspend(string SerialNumber, string PartNumber, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.Suspend(SerialNumber, PartNumber, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Resumes a suspended workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse Resume(string SerialNumber, string PartNumber, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.Resume(SerialNumber, PartNumber, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Cancels an active workflow
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse Cancel(string SerialNumber, string PartNumber, StatusEnum workflowDefinitionStatus = StatusEnum.Released)
            => new WorkflowResponse(_instance.Cancel(SerialNumber, PartNumber, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Adds unit in a boxbuild
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="ChildSerialNumber"></param>
        /// <param name="ChildPartNumber"></param>
        /// <param name="ActivityName"></param>
        /// <param name="inputValues"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse AddUnit(string SerialNumber, string PartNumber, string ChildSerialNumber, string ChildPartNumber, string ActivityName, Dictionary<string, object> inputValues, StatusEnum workflowDefinitionStatus)
            => new WorkflowResponse(_instance.AddUnit(SerialNumber, PartNumber, ChildSerialNumber, ChildPartNumber, ActivityName, inputValues, (napict.StatusEnum)(int)workflowDefinitionStatus));

        /// <summary>
        /// Removes a sub unit
        /// </summary>
        /// <param name="SerialNumber"></param>
        /// <param name="PartNumber"></param>
        /// <param name="ChildSerialNumber"></param>
        /// <param name="ChildPartNumber"></param>
        /// <param name="ActivityName"></param>
        /// <param name="inputValues"></param>
        /// <param name="workflowDefinitionStatus"></param>
        /// <returns></returns>
        public WorkflowResponse RemoveUnit(string SerialNumber, string PartNumber, string ChildSerialNumber, string ChildPartNumber, string ActivityName, Dictionary<string, object> inputValues, StatusEnum workflowDefinitionStatus)
            => new WorkflowResponse(_instance.RemoveUnit(SerialNumber, PartNumber, ChildSerialNumber, ChildPartNumber, ActivityName, inputValues, (napict.StatusEnum)(int)workflowDefinitionStatus));
    }
}
